packages/shared/src/components/checklist/SquadFirstCommentChecklistStep.tsx

```diff
@@ -19,7 +21,11 @@ const SquadFirstCommentChecklistStep = ({
         className="btn-primary"
         disabled={!welcomePost}
         onClick={() => {
-          router.push(`/posts/${welcomePost.id}?comment=Say hi 👋`);
+          router.push(
+            `/posts/${welcomePost.id}?comment=Hi my name is ${
+              user?.name || user.username
```

Comment:
Since user can be `undefined` We either need to do optional on both `user` getters or just remove them to expect user is always there. Like this its gonna break either way and its not clear what we want.

Reply:
Anonymous won't have checklist rendered right?

Reply:
Yeah, but if for ever reason if we render checklist for anonymous. Our typescript should be screaming that user could be undefined. Won't block but I think it's right from general logic perspective.

Reply:
Oh sorry so name is optional it can be empty on a user, but username is not.
Hope that actually answers it better.

We do expect user to be there always.

Reply:
Since this is going to be passed as part of the query param and I am unsure yet of the allowed special characters for the name, I think we should run it with `encodeURIcomponent`.

Reply:
Ah good point, we can do that for the name alone indeed 🙏
Username shouldn't allow it anyway